org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
Yeah, I missed that use case when improving type detection ;(

If we add

    _oidToPgName.put(arrayOid, pgArrayTypeName);

before

    pgArrayTypeName = "_" + pgTypeName;

it will be consistent as type[] format
Otherwise custom type detection needs to be improved and lookup internal name for arrays (should be easy to fix): https://github.com/pgjdbc/pgjdbc/blob/f3b0fd007ca2fc186626c4ee5157015932c4a359/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L300

we can probably look up both normal and array types in both cases.
I guess reverting to prefix _ is aligned with how it worked previously.

On the positive side, this change reduced type lookups for some scenarios ;)

Let me know if you want me to finish it up. And what have you decided as array format.